internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -33,69 +118,102 @@ func (connector *SingleUseConnector) Connect(
-			": %s", err)
-		connector.sendErrorToClient()
-		return nil, err
-	}
+	// Create the context for our connection
+	parentContext := context.Background()
+	ch := make(chan map[uint8][]byte)
 
-	return connector.backendConn, nil
-}
+	// Set a 'marker' for when the driver has finished connecting
+	driverConnSet := make(chan struct {})
 
-// ConnectToBackend establishes the connection to the target database and sets
-// the backendConnection field.
-func (connector *SingleUseConnector) ConnectToBackend(dataSourceName string) error {
-	var err error
+	// Add channels to the context, to retrive information from the driver
+	childContext := context.WithValue(parentContext, PreLoginResponseKey, ch)
 
-	// Using DSN (Data Source Name) string because gomssql forces us to.
-	//
-	// NOTE: Secretless has some unfortunate naming collisions with the
-	// go-mssqldb driver package.  The driver package has its own concept of a
-	// "connector", and its connectors also have a "Connect" method.
-	driverConnector, err := mssql.NewConnector(dataSourceName)
+	// Build a new driver connection
+	var driverConn driver.Conn
+
+	go func() {
+		// Kick off authentication through our third party connector
+		driverConn, err = driverConnector.Connect(childContext)
+		driverConnSet <- struct{}{}
+
+		if err != nil {
+			_, err = connector.sendErrorWithMessage("failed to connect to mssql server: %s", err)
+		}
+	} ()
+
+	// Blocks continuation until we've received the preLoginResponse from the driver
+	preloginResponse := <- ch
+
+	// We don't want the driver to use encryption when talking to the server
+	preloginResponse[mssql.PreloginENCRYPTION] = []byte{mssql.EncryptNotSup}
+
+	// Write the prelogin packet back to the user
+	err = mssql.WritePreloginWithPacketType(clientBuffer, preloginResponse, mssql.PackReply)
 	if err != nil {
-		connector.logger.Errorf("Failed to create a go-mssqldb connector: %s", err)
-		return err
+		return connector.sendErrorWithMessage("failed to write prelogin response: %s", err)
 	}
 
-	driverConn, err := driverConnector.Connect(context.Background())
+	// We actually don't need the client's handshake response;
+	// we just need for them to not be blocked
+	err = clientBuffer.ReadNextPacket()
 	if err != nil {
-		connector.logger.Errorf("failed to connect to mssql server: %s", err)
-		return err
+		return connector.sendErrorWithMessage("Failed to handle client prelogin request: %s", err)
 	}
 
+	// TODO: 	Send the login response to the client 	 (#1016)
+	// TODO: 	Verify appropriate errors are passed to the client (#1013)
+
+	// TODO: 	Replace this with an actual 'ok' message from the server
+	//			once login completes within the driver
+	if _, err = clientConn.Write(connector.CreateAuthenticationOKMessage()); err != nil {
+		return connector.sendErrorWithMessage("Failed to send a successful authentication"+
+			" response to the client"+
+			": %s", err)
+	}
+
+	// Block continuation until driver has completed connection
+	<- driverConnSet
```

Comment:
Fun aside:

I didn't see this until making my other comment above on renaming `drivingConnSet`,  but note how the comment is naturally using the language ("completed connection") that I suggested for the variable name.  This is a signal to lookout for: when your comments use different language from your variables, it's often a sign that the variable should be using the comment language.  Because our comments typically use the natural, informal words with which we really think.